packages/react-jss/src/withStyles.js

```diff
@@ -35,32 +36,20 @@ let managersCounter = 0
 
 const NoRenderer = (props: {children?: Node}) => props.children || null
 
-const getStyles = <Theme: {}>(styles: Styles<Theme>, theme: Theme) => {
+const getStyles = <Theme: {}>(styles: Styles<Theme>, theme: Theme, displayName: string) => {
   if (typeof styles !== 'function') {
     return styles
   }
-  if (process.env.NODE_ENV !== 'production') {
-    if (styles.length < 1) {
-      console.warn(
-        `
-        Warning: [JSS]: This Component uses themed styles notation (function) while not relying on a theme (0 arguments).
-        It slows your app down and you should rewrite these styles to plain object notation.
-      `
-          .replace(/[\s]{2,}/gim, ' ')
-          .trim()
-      )
-    }
-    if (styles.length > 1) {
-      console.warn(
-        `
-        Warning: [JSS]: This Component's themed styles expect more than 1 argument.
-        React-jss provides only one argument.
-      `
-          .replace(/[\s]{2,}/gim, ' ')
-          .trim()
-      )
-    }
-  }
+  warning(
+    styles.length < 1,
+    `Warning: [JSS]: <${displayName} />'s styles function doesn't rely on a theme. We recommend to rewrite it to plain object.\nRead more: https://github.com/cssinjs/jss/blob/master/docs/react-jss.md#basic`
```

Comment:
what if the styles function is used with multiple components?

Reply:
We have to discuss the format of the links we use and the content. I was actually thinking of separate documents, which explain exactly the error, not just generic documents. Basically, the way react does that. But its a bit out of scope for this issue.

Reply:
that doesn't change the misuse scenario in one or all of the components.

in my experience (few thousands components) people do it only when dont have time to split big stylesheet into separate ones and keep adding styles in one stylesheet for several components.

Reply:
yeah, could be good enough, I don't see a good other pointer except of the stack trace, fn.name won't be useful since its usually `styles`

Reply:
Why would it be a problem if it's used multiple times?

Reply:
Its just a bit of indirection, it might be used in multiple but will mention one. But its ok I guess.

Reply:
We need to get rid of the links, I don't want to have issues complaining about broken links. We need permalinks and full issue descriptions.

Reply:
If you don't mind I will change the error message, we have a separate issue for linking errors to some permlink with good deescriptions